src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+            preparedStatement.setLong(1, id);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                ShoppingCart cart = parseResult(sqlResult);
+                return Optional.of(cart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shopping cart by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private ShoppingCart parseResult(ResultSet sqlResult) throws SQLException {
+        Long id = sqlResult.getLong("shopping_cart_id");
+        Long userId = sqlResult.getLong("user_id");
+        List<Product> productList = new ArrayList<>();
+        do {
+            Product product = parseProduct(sqlResult);
+            if (null != product) {
+                productList.add(parseProduct(sqlResult));
+            }
+        } while (sqlResult.next());
+        return new ShoppingCart(id, userId, productList);
+    }
+
+    private Product parseProduct(ResultSet sqlResult) throws SQLException {
+        Long productId = sqlResult.getLong("product_id");
+        String name = sqlResult.getString("name");
+        Double price = sqlResult.getDouble("price");
+        if (name == null) {
+            return null;
+        }
+        return new Product(productId, name, price);
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart cart) {
+        String query = "DELETE FROM shopping_carts_products WHERE shopping_carts_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, cart.getId());
+            preparedStatement.executeUpdate();
+            insertProducts(cart, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't update cart for user = " + cart.getUserId(), e);
+        }
+        return cart;
+    }
+
+    private int insertProducts(ShoppingCart cart, Connection connection) throws SQLException {
+        String query = "INSERT INTO shopping_carts_products (shopping_carts_id, products_id) "
+                + "VALUES (?, ?)";
+
+        int result = 0;
+        for (Product product : cart.getProducts()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, cart.getId());
+            preparedStatement.setLong(2, product.getId());
+            result += preparedStatement.executeUpdate();
+        }
+        return result;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE shopping_carts"
+                + " SET deleted = true "
+                + "WHERE shopping_cart_id = ? and deleted = false";
+
+        String query2 = "DELETE FROM shopping_carts_products"
+                + " WHERE shopping_carts_id = ?";
```

Comment:
you are repeating your method "update" above - just call it instead of coding after you execute query 1

Reply:
ok. Done